iwyu_test_util.py

```diff
@@ -419,6 +436,112 @@ def _GetLaunchArguments(cc_file):
   return shlex.split(args)
 
 
+def _CheckForCompileTimeMacro(macro):
+    """This assumes that the local clang++ and the include-what-you-use
+    under test are using the same libclang, and therefore the same default libraries.
+
+    Look for the specified macro in the clang output.
+    """
+    txt = '#include <cstddef>'
+
+    result = subprocess.run(['clang', '-dM', '-E', '-x', 'c++', '-'], capture_output=True,
+                            input=txt, text=True)
+
+    return " "+macro+" " in str(result.stdout)
+
+
+def _CheckExtraCommandLine(args):
```

Comment:
`IWYU_EXTRA_ARGS` only represents arguments that are passed by whoever runs the test suite. I don't think that's enough, in general, don't you also want to inspect `// IWYU_ARGS` and any builtin hard-coded args? I.e. we should probably extract the argument building from `TestIwyuOnRelativeFile` and evaluate against the full arglist.